### ISSUE TITLE
[BR]: Several tests fail with Python 3.11

### ISSUE THREAD

**Issue by hobbes1069** [fail2ban/fail2ban#3260]
### Environment:

- Fail2Ban version  : 0.11.2
- OS, including release name/version : Fedora Rawhide (future f37)
- [x] Fail2Ban installed via OS/distribution mechanisms
- [ ] You have not applied any additional foreign patches to the codebase
- [ ] Some customizations were done to the configuration (provide details below is so)

### The issue:

Global inline flags (e.g. (?i)) can now only be used at the start of the regular expressions. Using them not at the start of expression was deprecated since Python 3.6. (Contributed by Serhiy Storchaka in bpo-47066.)

#### Any additional information

Full build logs, to jump to the tests search for %check:
https://hobbes1069.fedorapeople.org/build.log

#### Any customizations done to /etc/fail2ban/ configuration
The beginnings of a patch for Python 3.11:
```
Index: fail2ban-0.11.2/fail2ban/tests/actiontestcase.py
===================================================================
--- fail2ban-0.11.2.orig/fail2ban/tests/actiontestcase.py
+++ fail2ban-0.11.2/fail2ban/tests/actiontestcase.py
@@ -244,14 +244,14 @@ class CommandActionTest(LogCaptureTestCa
        setattr(self.__action, 'ab', "<ac>")
        setattr(self.__action, 'x?family=inet6', "")
        # produce self-referencing properties except:
-       self.assertRaisesRegexp(ValueError, r"properties contain self referencing definitions",
+       self.assertRaisesRegex(ValueError, r"properties contain self referencing definitions",
            lambda: self.__action.replaceTag("<a><b>",
                self.__action._properties, conditional="family=inet4")
        )
        # remore self-referencing in props:
        delattr(self.__action, 'ac')
        # produce self-referencing query except:
-       self.assertRaisesRegexp(ValueError, r"possible self referencing definitions in query",
+       self.assertRaisesRegex(ValueError, r"possible self referencing definitions in query",
            lambda: self.__action.replaceTag("<x"*30+">"*30,
                self.__action._properties, conditional="family=inet6")
        )
```

**Comment by sebres** [comment]
fixed in #3267

**Comment by sebres** [comment]
merged in 0.11 (c6cfd761bb90b4150181d292be3ce309af8cdd5b) and 1.0 (3c22ae161cfe23a9fb00cabb8a7de891e3e8edaf)
